paddle/operators/gather_func.h

```diff
@@ -0,0 +1,114 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#pragma once
+#include <cstring>
+#include "paddle/framework/tensor.h"
+#include "paddle/platform/place.h"
+#include "paddle/framework/ddim.h"
+
+/**
+ * Return a new tensor from source tensor, gathered according to index
+ * input[src]: type-T source Tensor
+ * input[Index]: type-int index Tensor (1-D)
+ * return: output tensor
+ */
+template <typename place, typename T>
+Tensor* Gather_func(Tensor* Src, Tensor* Index) {
+	// assert index is an int-type tensor?
+	// assert(Index->istype(int));
+
+	// check index of shape 1-D
+	assert(Index->dims().size()==1);
+	int index_size = Index->dims()[0];
+
+	// Source shape
+	auto src_dims = Src->dims();
```

Comment:
The indentation doesn't match Google C++ style. This error implies that you are not using pre-commit to automatically reformat your code.

Please run `pip install pre-commit` to install pre-commit. Then Paddle repo's configuration will enable Git to call it whenever you run `git commit`.